### PR TITLE
[CI] Add task UsePythonVersion@0 to Azure Pipelines

### PR DESCRIPTION
**Pull request by jmiaskox**
### Details:
 - *Use Python 3.10.7 from GitHub actions, allowing for easier upgrades in the future*

### Tickets:
 - *[CVS-100817](https://jira.devtools.intel.com/browse/CVS-100817)*

### CODE REVIEW

File: .ci/azure/linux.yml [hunk 1]
```diff
@@ -282,21 +300,23 @@ jobs:
...
       python3 -m pytest -s $(INSTALL_TEST_DIR)/pyngraph $(PYTHON_STATIC_ARGS) \
         --junitxml=$(INSTALL_TEST_DIR)/TEST-Pyngraph.xml \
         --ignore=$(INSTALL_TEST_DIR)/pyngraph/tests/test_onnx/test_zoo_models.py \
         --ignore=$(INSTALL_TEST_DIR)/pyngraph/tests/test_onnx/test_backend.py
-    env:
-      # because of static build libgna is needed for python binary
-      LD_LIBRARY_PATH: $(REPO_DIR)/temp/gna_03.00.00.1910/linux/x64
+    # env:
+    #   # because of static build libgna is needed for python binary
+    #   LD_LIBRARY_PATH: $(REPO_DIR)/temp/gna_03.00.00.1910/linux/x64
```
Comment (p-wysocki) [inline .ci/azure/linux.yml:313]: Should this be here?

File: .ci/azure/linux.yml [hunk 2]
```diff
@@ -305,16 +325,17 @@ jobs:
         --ignore=$(INSTALL_TEST_DIR)/pyopenvino/tests/test_onnx/test_backend.py
     env:
       # because of static build libgna is needed for python binary and mock_py frontend library
-      LD_LIBRARY_PATH: $(REPO_DIR)/temp/gna_03.00.00.1910/linux/x64:$(INSTALL_TEST_DIR)
+      # LD_LIBRARY_PATH: $(REPO_DIR)/temp/gna_03.00.00.1910/linux/x64:$(INSTALL_TEST_DIR)
```
Comment (p-wysocki) [inline .ci/azure/linux.yml:328]: Should this be here?

File: .ci/azure/linux.yml [hunk 3]
```diff
@@ -305,16 +325,17 @@ jobs:
...
 
   - script: |
+      export LD_LIBRARY_PATH=$(REPO_DIR)/temp/gna_03.00.00.1910/linux/x64:$(LD_LIBRARY_PATH)
       python3 -m pytest -s $(INSTALL_TEST_DIR)/mo/unit_tests --junitxml=$(INSTALL_TEST_DIR)/TEST-ModelOptimizer.xml
-    env:
-      # because of static build libgna is needed for python binary
-      LD_LIBRARY_PATH: $(REPO_DIR)/temp/gna_03.00.00.1910/linux/x64
+    # env:
+    #   # because of static build libgna is needed for python binary
+    #   LD_LIBRARY_PATH: $(REPO_DIR)/temp/gna_03.00.00.1910/linux/x64:$LD_LIBRARY_PATH
```
Comment (p-wysocki) [inline .ci/azure/linux.yml:338]: Should this be here?

**Review by p-wysocki — COMMENTED** [review]
There are some commented out lines, should they be there? There are more in different files, I only marked three occurrences.

File: .ci/azure/linux.yml (hunk 1 above)
Comment (jmiaskox) [reply .ci/azure/linux.yml]: Removed its not need anymore.

File: .ci/azure/android_arm64.yml [hunk 4]
```diff
@@ -62,8 +62,23 @@ jobs:
     TMP_DIR: /mnt/tmp
     SHARE_DIR: /mount/cinfsshare/onnxtestdata
     CCACHE_DIR: $(SHARE_DIR)/ccache/master/android_arm64
+    LD_LIBRARY_PATH: $(Agent.ToolsDirectory)/Python/$(OV_PYTHON_VERSION)/x64/lib
+    OV_PYTHON_VERSION: 3.10.7
```
Comment (ilya-lavrenov) [inline .ci/azure/android_arm64.yml:66]: see warnings https://dev.azure.com/openvinoci/dldt/_build/results?buildId=478842&view=results

Comment (p-wysocki) [reply .ci/azure/android_arm64.yml:66]: @jmiaskox is it something we can fix? I remember that there were some issues with `major.minor` version format before.

Comment (jmiaskox) [reply .ci/azure/android_arm64.yml:66]: Well this way we can specify exact version of Python, It is recommended to specify only major or major and minor version. Overall its only warning info that not should affect builds.

**Review by p-wysocki — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]

**Pull request merged**